nuxt.config.js

```diff
@@ -1,87 +1,80 @@
-/** @type {import('highcharts/highcharts').Options} */ 
-const setOptions = { 
-  lang: { // <-- we correctly get intellisense
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable no-unused-vars */
+/** @type {import('highcharts/highcharts').Options} */
+const setOptions = {
+  lang: {
+    // <-- we correctly get intellisense
     // decimalPoint: '.' // this can be changed to anything, like ',' and it works
   }
-}
+};
 
 /** @type {import('highcharts/highcharts').Options} */
-const chartOptions = { // <-- we also get intellisense correctly. 
+const chartOptions = {
+  // <-- we also get intellisense correctly.
   credits: {
     enabled: true // <-- typing "e" suggests "enabled?" ok
   }
-}
+};
 // Important note: as of 12/2020 it seems...
 // per the API docs, the "setOptions" get passed to the "Highcharts.setOptions" method
 // while the chartOptions would get passed to the *Highcharts.chart* method
 
-module.exports = {
-  env: {},
-  mode: 'spa', // 'universal', // TBD: use highcharts *src* for ssr mode?
-  /*
-   ** Headers of the page
-   */
+export default {
+  // Global page headers (https://go.nuxtjs.dev/config-head)
   head: {
-    title: process.env.npm_package_name || '',
+    title: "highcharts",
     meta: [
-      { charset: 'utf-8' },
-      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
-      {
-        hid: 'description',
-        name: 'description',
-        content: process.env.npm_package_description || ''
-      }
+      { charset: "utf-8" },
+      { name: "viewport", content: "width=device-width, initial-scale=1" },
+      { hid: "description", name: "description", content: "" }
     ],
-    link: [{ rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' }]
+    link: [{ rel: "icon", type: "image/x-icon", href: "/favicon.ico" }]
   },
-  /*
-   ** Customize the progress-bar color
-   */
-  loading: { color: '#fff' },
-  /*
-   ** Global CSS
-   */
-  css: ['~/assets/main.css'],
-  /*
-   ** Plugins to load before mounting the App
-   */
+  // target: 'static', // comment out for static else it will be SSR
+  // Global CSS (https://go.nuxtjs.dev/config-css)
+  css: ["~/assets/main.css"],
+
+  // Plugins to run before rendering page (https://go.nuxtjs.dev/config-plugins)
   plugins: [],
-  /*
-   ** Nuxt.js dev-modules
-   */
-  buildModules: [],
-  /*
-   ** Nuxt.js modules
-   */
+
+  // Auto import components (https://go.nuxtjs.dev/config-components)
+  components: true,
+
+  // Modules for dev and build (recommended) (https://go.nuxtjs.dev/config-modules)
+  buildModules: [
+    // https://go.nuxtjs.dev/typescript
+    "@nuxt/typescript-build"
```

Comment:
Do you have insight as to why the "require.context" error goes away in SSR-mode when using the typescript-build? I think it would be helpful to know not just here, but I think some folks ask about it on stackoverflow and other projects. (i.e., it's not a highcharts-specific issue, and others could probably be helped by the answer)

Reply:
i am not 100% sure but
- could be that using nuxt TS actually figures out the type of `require` in this particular function ?

Reply:
Hmm. for some reason, I still encounter the warning, even with the ts build. I'll take some time to investigate the issue. I know the block of code causing it, I just need to troubleshoot ways to mute the warning (https://github.com/webpack/webpack/issues/2675 might give some insight)

Reply:
Ok, I was able to mute the warning and get this fixed. The key was to mock out the require.context.keys() method not require.context, which is what we want from our tests anyway.